packages/pds/src/repo/types.ts

```diff
@@ -42,6 +42,19 @@ export type PreparedDelete = {
   swapCid?: CID | null
 }
 
+export type CommitOp = {
+  action: 'create' | 'update' | 'delete'
+  path: string
+  cid: CID | null
+  prev?: CID
+}
```

Comment:
Wonder if it could be useful for correctness to discriminate on `action`, e.g.:
```ts
export type CommitOp =
  | {
      action: 'create'
      path: string
      cid: CID
      prev: null
    }
  | {
      action: 'update'
      path: string
      cid: CID
      prev: CID
    }
  | {
      action: 'delete'
      path: string
      cid: null
      prev: CID
    }
```

Reply:
It might be more correct, but it leads to really annoying verbose code in `repo/transactor`

This type is a direct reflection of the type in the firehose stream, and this data is only ever used in that context so I feel pretty comfortable about it